src/Uno.UWP/UI/Core/CoreDispatcher.cs

```diff
@@ -159,8 +159,6 @@ public UIAsyncOperation RunAsync(CoreDispatcherPriority priority, DispatchedHand
 		/// <remarks>Can only be invoked on the UI thread</remarks>
 		internal UIAsyncOperation RunAsync(CoreDispatcherPriority priority, CancellableDispatchedHandler handler)
 		{
-			CoreDispatcher.CheckThreadAccess();
```

Comment:
I vaguely recall that this was here because there's a potential race condition if it's called off the UI thread... did you figure out why this was here and fix the problem?

Reply:
Humm good question! So it has been introduced by https://github.com/unoplatform/uno/commit/eb15bcaa to avoid a race condition (according to the commit details). And indeed in the previous implementation:
```cs
internal UIAsyncOperation RunAsync(CoreDispatcherPriority priority, CancellableDispatchedHandler handler)
{
	CoreDispatcher.CheckThreadAccess();

	UIAsyncOperation operation = null;

	DispatchedHandler nonCancellableHandler = () => handler(operation.Token);

	return operation = EnqueueOperation(priority, nonCancellableHandler);
}
```
the `operation` might be `null` if the `nonCancellableHandler` is invoked really quickly. So it will fail when it tries to access to the `operation.Token`.

The current implementation avoids that as the `operation` is now set by the `EnqueueOperation` **before** it actually enqueues the `nonCancellableHandler` on the dispatcher.